libraries/botbuilder-ai/tests/qnaMakerDialog.test.js

```diff
@@ -52,13 +98,159 @@ describe('QnAMakerDialog', function() {
             const NOT_V5_HOSTNAME = 'myqnainstance.net/qnamaker';
     
             // Missing authority
-            const noAuthority = new QnAMakerDialog(KB_ID, ENDPOINT_KEY, NOT_V5_HOSTNAME);
+            const noAuthority = new QnAMakerDialog(kbId, endpointKey, NOT_V5_HOSTNAME);
             const noAuthorityClient = await noAuthority.getQnAClient({state: {}});
     
             ok(noAuthorityClient instanceof QnAMaker);
-            strictEqual(noAuthorityClient.endpoint.knowledgeBaseId,  KB_ID);
-            strictEqual(noAuthorityClient.endpoint.endpointKey, ENDPOINT_KEY);
+            strictEqual(noAuthorityClient.endpoint.knowledgeBaseId,  kbId);
+            strictEqual(noAuthorityClient.endpoint.endpointKey, endpointKey);
             strictEqual(noAuthorityClient.endpoint.host, createHostName(NOT_V5_HOSTNAME));
         });
+        
+        it('should log telemetry that includes question and username if logPersonalInformation is true in env file', async () => {
```

Comment:
Something to consider - Sinon (in my opinion) offers a much more expressive and robust option for registering expectations and mocks. I've been rewriting several tests using this approach. It's handy as you only have to express the things you expect without needing to be as careful about replicating any functionality.

[Here](https://github.com/microsoft/botbuilder-js/pull/2912/files#diff-df984573d25004aa79b9c21819010fb506d47ef2e3bf8b0515006737c082ff32R79) is an example of registering expectations on a method that is eventually invoked by some other class entirely.

If you create a sandbox, all you need to do at the end of the test is to call `sandbox.verify()` to ensure that all the mocks/expectations you registered were called in all the ways you expected.

You don't have to take this approach, but it can be a big productivity boost, so I encourage you to look into it.

Reply:
I believe you were already aware of this, but @mdrichardson FYI